internal/rest/getv2clusters_test.go

```diff
@@ -532,7 +542,7 @@ func TestGetV2Clusters200(t *testing.T) {
 				name: "filtered clusters by providerStatus",
 				clusters: []capi.Cluster{
 					generateClusterWithStatus(ptr("example-cluster-1"), ptr("v1.30.6+rke2r1"), clusterStatusReady),
-					generateClusterWithStatus(ptr("example-cluster-2"), ptr("v1.20.4+rke2r1"), clusterStatusInProgressControlPlane),
+					generateClusterWithStatus(ptr("example-cluster-2"), ptr("v1.20.4+rke2r1"), clusterStatusFailed),
```

Comment:
this change was needed because in progress results in a `not ready` message which matches the `ready` query.

the `generateClusterInfo` function needs to be updated to handle the in progress state, not planned for this pr.

Reply:
what about `clusterStatusInProgressControlPlane`?